doc/tutorial/statistical_inference/model_selection.rst

```diff
@@ -41,42 +41,60 @@ data in *folds* that we use for training and testing::
     >>> print(scores)
     [0.93489148580968284, 0.95659432387312182, 0.93989983305509184]
 
-.. currentmodule:: sklearn.cross_validation
+.. currentmodule:: sklearn.model_selection
 
-This is called a :class:`KFold` cross validation
+This is called a :class:`KFold` cross-validation.
 
 .. _cv_generators_tut:
 
 Cross-validation generators
 =============================
 
+Scikit-learn has a collection of classes which can be used to generate lists of
+train/test indices for popular cross-validation strategies.
 
+They expose a ``split`` method which accepts the input
+dataset to be split and yields the train/test set indices for each iteration
+of the chosen cross-validation strategy.
 
-The code above to split data in train and test sets is tedious to write.
-Scikit-learn exposes cross-validation generators to generate list
-of indices for this purpose::
+This example shows an example usage of the ``split`` method.
 
-    >>> from sklearn import cross_validation
-    >>> k_fold = cross_validation.KFold(n=6, n_folds=3)
-    >>> for train_indices, test_indices in k_fold:
+    >>> from sklearn.model_selection import KFold, cross_val_score
+    >>> X = ["a", "a", "b", "c", "c", "c"]
+    >>> k_fold = KFold(n_folds=3)
+    >>> for train_indices, test_indices in k_fold.split(X):
     ...      print('Train: %s | test: %s' % (train_indices, test_indices))
     Train: [2 3 4 5] | test: [0 1]
     Train: [0 1 4 5] | test: [2 3]
     Train: [0 1 2 3] | test: [4 5]
 
-The cross-validation can then be implemented easily::
+The cross-validation can then be performed easily::
 
-    >>> kfold = cross_validation.KFold(len(X_digits), n_folds=3)
+    >>> kfold = KFold(n_folds=3)
     >>> [svc.fit(X_digits[train], y_digits[train]).score(X_digits[test], y_digits[test])
-    ...          for train, test in kfold]
+    ...          for train, test in k_fold.split(X_digits)]
     [0.93489148580968284, 0.95659432387312182, 0.93989983305509184]
 
-To compute the ``score`` method of an estimator, the sklearn exposes
-a helper function::
+This can be directly calculated using the ``cross_val_score`` helper. Given an
+estimator, the cross-validation object and the input dataset, it splits the
+data into train/test set, trains the estimator using the training set and
```

Comment:
I would add "repeatedly" so that it reads "splits the data repeatedly into a training and a test set" ...|

Currently that is not clear. Then I would split the sentence in two.